### PR TITLE
Actually made gear harness properly adjustable

### PR DESCRIPTION
**Pull request by Putnam3145**
## About The Pull Request

the code is bad and this fixes my earlier fix. This one I actually tested properly instead of just assuming the code is sane.

## Why It's Good For The Game

i like non-error'd things.

## Changelog
:cl:
fix: gear harness alt-click is now sane
code: rolldown() and toggle_jumpsuit_adjust() now no longer mix behavior-that-should-be-overridden and behavior-that-shouldn't-be-overridden in ways that make no sense.
/:cl:

### CODE REVIEW

**Review by DeltaFire15 — CHANGES_REQUESTED** [review]
Please do not override rolldown(), modify toggle_jumpsuit_adjust() instead.

Reason for that is rolldown() containing a bunch of checks, like if you are even allowed to do it, in the parent proc.

**Comment by Putnam3145** [comment]
> 
> 
> Please do not override rolldown(), modify toggle_jumpsuit_adjust() instead.
> 
> Reason for that is rolldown() containing a bunch of checks, like if you are even allowed to do it, in the parent proc.

it also includes a bunch of hardcoded nonsense that doesn't make sense in this context in the first place because it has a fuckton of assumptions of what alt-clicking is supposed to mean in the first place and the fact that I was using it is (as you can plainly see) the reason it was breaking in the first place.

**Comment by Putnam3145** [comment]
@DeltaFire15 alright, I've now reworked rolldown() and toggle_jumpsuit_adjust() so that you can actually customize jumpsuit adjusting instead of clearly being written with the assumption that neither would ever be overridden, as they were. I don't know why I didn't do this in the first place.

File: code/modules/clothing/under/miscellaneous.dm [hunk 1]
```diff
@@ -172,11 +172,13 @@
...
-	if(adjusted)
-		body_parts_covered = NONE
-	else
+	if(!body_parts_covered)
+		to_chat(usr, "<span class='notice'>Your gear harness is now covering your chest and groin.</span>")
 		body_parts_covered = CHEST|GROIN
+	else
+		to_chat(usr, "<span class='notice'>Your gear harness is no longer covering anything.</span>")
+		body_parts_covered = NONE
+	return !body_parts_covered
```
Comment (DeltaFire15) [inline code/modules/clothing/under/miscellaneous.dm]: This line should probably always return true considering the return value is used to see if the body needs to be updated.

Comment (Putnam3145) [reply code/modules/clothing/under/miscellaneous.dm]: Also done.

**Pull request merged**